### DISCUSSION TITLE
Is there an official list of supported browsers?

### DISCUSSION THREAD

**Discussion by jannisg** [pmndrs/jotai#788] [Q&A] [answered]
The readme doesn't mention it and I could not find it in the official documentation site either.

The closest I could find was this [comment on a GitHub issue](https://github.com/pmndrs/jotai/issues/48#issuecomment-815374467) that mentions that it _should_ work provided one polyfills `Map`, `Set` and `Symbol`.

Context: My organisation still needs to support IE11 for a bit longer and we are currently evaluating atom-based state management libraries to move away from redux.

**Comment by dai-shi** [comment, marked as answer]
In terms of transpilation, our ESM build is for modern browsers, and our CJS build is traditional browsers.
But, I don't think we can list supported browsers only from it...
The comments in #48 still hold. It uses `Map`, `Set`, `Symbol` and `WeakMap`, which may not work on old browsers.
If IE11 support is a requirement, my suggestion is [zustand](https://github.com/pmndrs/zustand), which is not atom based, though.

**Reply by jannisg** [reply]
Thanks for the reply :)

So far with the polyfills, jotai seems to work as expected though I have yet to test some of its patterns in IE (Provider, useHydrateAtom, etc) that make use of those newer features.

I guess the tricky bit will be staying up to date when jotai may introduce new requirements in minor versions that could break IE11, or would you release those types of changes as a new major version?

I'm guessing since IE11 is not supported they could just be minor bumps technically.

**Reply by dai-shi** [reply]
It's good to hear that the polyfills work.

And, yes, your assumption is correct. IE11 is not officially supported, minor bumps may break technically.

**Reply by jannisg** [reply]
Yep, fair enough :) 

Just as an FYI, I think `WeakSet` would also need to be polyfilled, see: https://github.com/pmndrs/jotai/blob/b98f865b9f1342dcf61f88cb282fa3994b827449/src/utils/useHydrateAtoms.ts#L37
